### DISCUSSION TITLE
Follow up question about #2577 (alternative fix)

### DISCUSSION THREAD

**Discussion by dnfeitosa** [MorphiaOrg/morphia#2597] [General]
@evanchooly thank you for the quick response and fix for #2577! I have a follow-up question about it.

Before you released the fix in 2.4.6, I got the problem solved with a custom `HashMapCodec` that gets initialized using a different provider and propagates the complete registry with the additional custom codecs. It is overriding the instance installed by [MorphiaTypesCodecProvider](https://github.com/MorphiaOrg/morphia/blob/master/core/src/main/java/dev/morphia/mapping/codec/MorphiaTypesCodecProvider.java#L35
).

Here's the custom codec and provider:

```
public class HashMapCodecProvider implements CodecProvider {

  @Override
  public <T> Codec<T> get(Class<T> clazz, CodecRegistry registry) {
    if (HashMap.class.isAssignableFrom(clazz)) {
      return (Codec<T>) new HashMapCodec(registry);
    }
    return null;
  }
}

public class HashMapCodec extends MapCodec {

  /**
   * Do not delete this constructor! It ensures the codec registries from the MongoDB driver are properly propagated. 
   *
   * @param registry Complete codec registry from MongoDB's driver.
   */
  public HashMapCodec(CodecRegistry registry) {
    super(registry);
  }

  @Override
  public Class<Map<String, Object>> getEncoderClass() {
    return (Class<Map<String, Object>>) ((Class) HashMap.class);
  }
}
```

So far, it has worked fine with the models from my projects (even the models with the most _unusual_ design choices), and I'm planning to keep it around, but I wanted to rely on your experience and confirm this is not going to introduce other issues down the line.

Are there any obvious problems that you can think of about using this codec?

Thank you for your hard work keeping Morphia!

** Please complete the following information: **
Server Version: 5.0.21
Driver Version: Java 4.10.2
Morphia Version: 2.4.6

**Comment by evanchooly** [comment]
It's *probably* fine. If things get weird just remember it's there so you can investigate it.  HashMapCodec relies on MapCodec from the driver which is constructed oddly. I'll probably just write a new codec for it just to be safe.
